cwltool/draft2tool.py

```diff
@@ -414,10 +416,12 @@ def collect_output(self, schema, builder, outdir):
                                 files["contents"] = contents
                             filesize = 0
                             while contents != "":
-                                checksum.update(contents)
+                                if compute_checksum:
+                                    checksum.update(contents)
                                 filesize += len(contents)
                                 contents = f.read(1024*1024)
```

Comment:
If we're not bothering to compute a checksum, this whole "read all the data" loop seems like a really expensive way to determine file size. Is there a more efficient way to get file size?